forms/gridfield/GridFieldAddExistingAutocompleter.php

```diff
@@ -214,6 +214,17 @@ public function doSearch($gridField, $request) {
 			->sort(strtok($searchFields[0], ':'), 'ASC')
 			->limit($this->getResultsLimit());
 
+		$belongs_to = Config::inst()->get($dataClass, 'belongs_to');
+		if( !empty($belongs_to) ) {
+			foreach ($searchFields as $searchField) {
+				if( $relation = strtok($searchField,'.') ) {
+					if (in_array($relation, $belongs_to)) {
+						$results = $results->leftJoin($relation, "{$relation}.{$dataClass}ID={$dataClass}.ID");
```

Comment:
Please ANSI quote your join condition. :D

`"\"{$relation}\".\"{$dataClass}ID\" = \"{$dataClass}\".\"ID\""`

Also, $dataClass isn't the correct key; You need to look up the relation name on the other end.

E.g. a belongs_to will have a has_one on the other side, and the name of the has_one will determine the database field to join on.

You can use DataObject::getRemoteJoinField to get this.